Dockerfile

```diff
@@ -0,0 +1,70 @@
+# syntax = docker/dockerfile:1
+ARG NODE_VERSION=20.0.0
+ARG SECRET_KEY
+ARG APP_DEBUG
+ARG TEST_SALEOR_API_URL
+FROM node:${NODE_VERSION}-slim as base
+
+# Adjust NODE_VERSION as desired
+
+ENV NODE_VERSION="$NODE_VERSION"
+
+
+LABEL fly_launch_runtime="Next.js"
+
+# Next.js app lives here
+WORKDIR /app
+
+# Install pnpm
+ARG PNPM_VERSION=8.12.0
+RUN npm install -g pnpm@$PNPM_VERSION
+
+
+# Throw-away build stage to reduce size of final image
+FROM base as build
+ARG SECRET_KEY
+ARG APP_DEBUG
+ARG TEST_SALEOR_API_URL
+ARG UPSTASH_URL
+ARG UPSTASH_TOKEN
+ARG APL
+
+# Set these via `docker run -e APP_DEBUG=debug ...`
+# OR set them in your deployed containers environment secrets
+# 
+# ENV TEST_SALEOR_API_URL="$TEST_SALEOR_API_URL"
+# ENV SECRET_KEY="$SECRET_KEY"
+# ENV APP_DEBUG="$APP_DEBUG"
+# ENV UPSTASH_URL="$UPSTASH_URL"
+# ENV UPSTASH_TOKEN="$UPSTASH_TOKEN"
+# ENV APL="${APL-upstash}"
+ENV NODE_ENV="production"
+
+# Install packages needed to build node modules
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y build-essential node-gyp pkg-config python-is-python3
+
+# Install node modules
+# Copy application code
+WORKDIR /app
+COPY . .
+RUN npm i -g graphql typescript @graphql-codegen/cli --legacy-peer-deps --force
+RUN pnpm install @saleor/json-schema-compiler
+RUN pnpm install --frozen-lockfile --prod=false
+
+# # # Build application
+RUN pnpm run build
```

Comment:
Then, you can copy the source code and rebuild it:

```suggestion
# # # Build application
COPY . .
RUN pnpm run build
```

Though, I recommend to only copy what's necessary in order to avoid unnecessary rebuilds and unneeded files

Reply:
updated